### ISSUE TITLE
ImageStim (transparent PNG) not displayed correctly

### ISSUE THREAD

**Issue by carolfs** [psychopy/psychopy#2390]
The following image is not displayed correctly as an ImageStim on macOS High Sierra.

![ring1](https://user-images.githubusercontent.com/55320/55638976-ad0e9300-57c8-11e9-88a3-a43609c68bc9.png)

Here is the code used to display it:

```
from psychopy import visual, core

win = visual.Window(size=(800, 600), color=(1, 1, 1), fullscr=False)
ring_img = visual.ImageStim(
    win,
    image='ring1.png',
)
ring_img.draw()
win.flip()
core.wait(5)
```

And here is how it is displayed:

![Screen Shot 2019-04-05 at 17 31 08](https://user-images.githubusercontent.com/55320/55638970-a6801b80-57c8-11e9-83c6-4cfd8c73d335.png)

It looks like the image is getting wrapped and the top pixels are displayed at the bottom.

I'm running PsychoPy 3.0.7 and pygame 1.9.5.

**Comment by m-macaskill** [comment]
Thanks for providing a nice minimal working example. 

I can't replicate this visual glitch on Mac OS Mojave 10.14.4, PsychoPy 3.1, however. What happens if you attempt this on another computer/OS?

**Comment by carolfs** [comment]
It looks different, but still bad on Windows 10 Home, PsychoPy 3.0.0b11.

![ss](https://user-images.githubusercontent.com/55320/58084749-bf4b6180-7bbb-11e9-8063-ecb2482b42be.png)

I found a workaround for this issue: adding a 1px transparent margin around the image.

**Comment by m-macaskill** [comment]
I'm glad you found a work-around. That you see the issue on two different operating systems, but it is not reproducible for me, makes me wonder if this is an issue that was resolved in a PsychoPy update. The software was in constant flux at the 3.0.0 release, and you are using one of the many betas in that series. I wonder what would happen if you try the latest release, currently 3.1.2, and go back to the original image file?

Also, lastly as a stab in the dark, I wonder if the window is being created with the old, deprecated `pygame` window type instead of the more current `pyglet`. According to the API, if `winType` is not specified, then PsychoPy will revert to user/site preferences, which might be set to `pygame`? https://www.psychopy.org/api/visual/window.html#psychopy.visual.Window

i.e. 

win = visual.Window(winType = 'pyglet', size=(800, 600), color=(1, 1, 1), fullscr=False)

**Comment by carolfs** [comment]
I added `winType='pyglet'` to the code, uninstalled `pygame`, upgraded PsychoPy to 3.1.2, deleted the `~/.psychopy3` directory. The results are exactly the same. Is there any other place where old config files might be stored?

**Comment by peircej** [comment]
I think this might be a simpler matter of the image being drawn just off a pixel and the graphics card. Mike has a retina display (the pixels are tiny so there won't be any visible issues).

By default images use interpolate=False (which is nearest neighbour interpolation) but if the nearest neighbour is not a pixel then nothing will be drawn. Two ways to test this: set you image to use (linear) interpolation:
```python
ring_img = visual.ImageStim(
    win,
    image='ring1.png',
    interpolate=True,
)
```
(maybe this should actually be the default)

Test the effect of this across space (as the image moves in and out of whole pixels):
``` python
from psychopy import visual, core

win = visual.Window(size=(800, 600), color=(1, 1, 1), fullscr=False)
ring_img = visual.ImageStim(
    win,
    image='ring1.png',
    units='pix',
    interpolate=False,
)
print(ring_img.interpolate)
for frameN in range(100):
    ring_img.pos += 0.1
    ring_img.draw()
    win.flip()
core.wait(5)
```

**Comment by carolfs** [comment]
With `interpolate=True`, the drawing still looks wrong to me:
![Screen Shot 2019-06-28 at 17 38 31](https://user-images.githubusercontent.com/55320/60354411-7497e300-99cc-11e9-9892-b58f61806753.png)

With `interpolate=False`, it alternates between looking good and bad.

**Comment by peircej** [comment]
OK. Thanks for confirming my intuitions about what must be happening. From this info I think I'll be able to fix that and get back to you.

**Comment by peircej** [comment]
I'm pretty sure this was indeed fixed in July 2019 in commits 389e1571658ac and d2a9eddb203 and released in PsychoPy 3.2.0
Sorry I failed to come and feedback that info here earlier! 😄
